src/prefect/flows.py

```diff
@@ -1812,61 +1812,133 @@ def my_other_flow(name):
             serve(hello_deploy, bye_deploy)
         ```
     """
-    from rich.console import Console, Group
-    from rich.table import Table
 
     from prefect.runner import Runner
 
+    try:
+        asyncio.get_running_loop()
+        raise RuntimeError(
+            "Cannot call `serve` in an asynchronous context. Use `aserve` instead."
+        )
+    except RuntimeError as exc:
+        if "no running event loop" not in str(exc):
+            raise
```

Comment:
We have an internal `is_in_async_context` util that can simplify this:
```suggestion
    if is_in_async_context:
        raise RuntimeError(
            "Cannot call `serve` in an asynchronous context. Use `aserve` instead."
        )
```
You can import this util like this:
```python
from prefect._internal.compatibility.async_dispatch import is_in_async_context